lib/sinatra/base.rb

```diff
@@ -358,6 +358,7 @@ def send_file(path, opts = {})
       result    = file.serving env
       result[1].each { |k,v| headers[k] ||= v }
       headers['Content-Length'] = result[1]['Content-Length']
+      headers['Content-Type'] = result[1]['Content-Type']
```

Comment:
2 years late, but if we follow ^, isn't that's what being done in the `result[1].each {...}` block?